pyscf_ipu/experimental/basis.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+from typing import Tuple
+
+import chex
+import jax.numpy as jnp
+
+from .orbital import Orbital
+from .structure import Structure
+from .types import FloatN, FloatNx3, FloatNxM
+
+
+@chex.dataclass
+class Basis:
+    orbitals: Tuple[Orbital]
+    structure: Structure
```

Comment:
[Probably not for this PR but...] Duplicated information here, as atom positions are both in orbitals and structure - how far coudl we get if we explicitly convert from Structure -> Basis and don't maintain the handy back-pointer?

Reply:
I have thought about removing the `structure` field from the `Basis` class as you point out the atom positions are redundantly stored with in the orbital / primitives.  The trouble is we need to know the structure for computing the occupied orbitals.  Definitely needs some more thought!